### PR TITLE
Feature/case data card

### LINKED ISSUE #262: [EPIC]: Add Case Data Card to Scenario Cards
### Related Issues/Tasks:

- [ ] #294 

---
### Details
The goal is to display the Case data with its own Scenario Card (black) in the scenario cards section.
This way case data for a relevant day can be read outside the tooltip in the line chart.

---
### Questions

- How to display scenario cards with no data (date out of range)?
  > Flip card (inactive) -> add "N/A" to card back

- Make it possible to select dates outside of scenario time ranges?
  > Yes

- How to handle the play function?
  > Start on selectd date, go until end of data

- Case data card selectable to display values on district map?
  > Yes

---

### PR DESCRIPTION
**Pull request by JonasGilg**
### Description

Adds a new card for case data in black. Additionally the Map now also can show case data if the case data card is selected.

The selected compartment is now also highlighted in the cards.

A little bit of refactoring also made it into this PR.

#### Related Issues

Closes #262

### Checklist

**I, the author of this PR checked the following requirements for good software quality:**

- [x] The code is properly formatted (I ran the formatter)
- [x] The code is written with our software quality standards (I ran the linter)
- [x] The code is written using our code style
- [x] Extensive in source documentation has been added
- [ ] Unit and/or integration tests have been added
- [x] All texts have been internationalized with at least the following languages:
  - [x] English
  - [x] German
- [x] I tried addressing all new accessibility problems displayed in the console and documented if they can't be fixed
- [x] I attached performance measurements to prevent performance degradation
- [x] I added the changes to the next release section of the [changelog](../frontend/docs/changelog/changelog-en.md)

**I, the reviewer checked the following things:**

- [ ] I ran the software once and tried all new and related functionality to this PR
- [ ] I looked at all new and changed lines of code and commented on possible problems
- [ ] I read the added documentation and checked if it is understandable and clear
- [ ] I checked the added tests for completeness
- [ ] I checked the internationalized strings for spelling errors
- [ ] I checked the performance metrics for problems or unexplained degradation
- [ ] I checked that the changes are noted in the [changelog](../frontend/docs/changelog/changelog-en.md)

### CODE REVIEW

File: frontend/src/components/Scenario/CaseDataCard.tsx [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+      day: day ?? '',
+      groups: ['total'],
+    },
+    {skip: !day}
+  );
+
+  const compartmentValues = useMemo(() => {
+    if (data && data.results.length > 0) {
+      return data.results[0].compartments;
+    }
```
Comment (JonasGilg) [inline frontend/src/components/Scenario/CaseDataCard.tsx:46]:
```suggestion
    if (data || data.results.length > 0) {
      return data.results[0].compartments;
    }
```

File: frontend/src/components/Scenario/index.tsx [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+          fullWidth={true}
+          open={open}
+          onClose={() => {
+            if (groupEditorUnsavedChanges) {
+              setCloseDialogOpen(true);
+            } else {
+              setOpen(false);
+            }
+          }}
+        >
```
Comment (NXXR) [inline frontend/src/components/Scenario/index.tsx:108]:
```suggestion
        <Dialog
          id='manage-filters-dialog'
          maxWidth='lg'
          fullWidth={true}
          open={open}
          onClose={() => {
            if (groupEditorUnsavedChanges) {
              setCloseDialogOpen(true);
            } else {
              setOpen(false);
            }
          }}
        >
```

File: frontend/src/components/Scenario/CompartmentList.tsx [hunk 3]
```diff
@@ -0,0 +1,243 @@
...
+            compartments.findIndex((o) => {
+              return o === selectedCompartment;
+            }) > 4
+          ) {
+            if (compartments.length > 0) {
+              dispatch(selectCompartment(compartments[0]));
+            }
+          }
+        }}
+      >
```
Comment (NXXR) [inline frontend/src/components/Scenario/CompartmentList.tsx:111]:
```suggestion
      <Button
        id='toggle-expanded-compartments-button'
        variant='outlined'
        color='primary'
        sx={{
          margin: theme.spacing(3),
          marginTop: theme.spacing(4),
          marginBottom: 0,
          padding: theme.spacing(1),
        }}
        aria-label={t('scenario.more')}
        onClick={() => {
          dispatch(toggleCompartmentExpansion());
          if (
            compartments.findIndex((o) => {
              return o === selectedCompartment;
            }) > 4
          ) {
            if (compartments.length > 0) {
              dispatch(selectCompartment(compartments[0]));
            }
          }
        }}
      >
```

File: frontend/src/components/Scenario/CompartmentList.tsx [hunk 4]
```diff
@@ -0,0 +1,243 @@
...
+    </Box>
+  );
+}
+
+/** This component renders the simulation start date together with a descriptive label. */
+function SimulationStartTitle(): JSX.Element {
+  const theme = useTheme();
+  const {t, i18n} = useTranslation();
+
+  const startDay = useAppSelector((state) => state.dataSelection.minDate);
```
Comment (NXXR) [inline frontend/src/components/Scenario/CompartmentList.tsx:123]: Not entirely sure if this works properly.
Day needs to be adjusted.
```suggestion
  const startDay = useAppSelector((state) => {
    let date = new Date(state.dataSelection.minDate);
    // start day is the day before the first simulation result
    date = new Date(date.getDate() - 1);
    return `${date.toISOString()[0:9]}`
  });
```

File: frontend/src/components/Scenario/DataCardList.tsx [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+
+      //activate all scenarios initially
+      if (!activeScenarios) {
+        scenarios.forEach((scenario) => {
+          dispatch(toggleScenario(scenario.id));
+        });
+      }
+
+      if (scenarios.length > 0) {
+        // It seems, that the simulation data is only available from the second day forward.
```
Comment (NXXR) [inline frontend/src/components/Scenario/DataCardList.tsx:62]:
```suggestion
        // The simulation data (results) are only available one day after the start day onward.
```

File: frontend/src/components/Scenario/DataCard.tsx [hunk 6]
```diff
@@ -0,0 +1,394 @@
...
+          flexGrow: 0,
+          flexShrink: 0,
+          width: '200px',
+          boxSizing: 'border-box',
+          marginX: '2px',
+          marginY: theme.spacing(2),
+          marginBottom: 0,
+        }}
+        onMouseLeave={() => setHover(false)}
+      >
```
Comment (NXXR) [inline frontend/src/components/Scenario/DataCard.tsx:139]:
```suggestion
      <Box
        id='scenario-card-container'
        sx={{
          position: 'relative',
          zIndex: 0,
          flexGrow: 0,
          flexShrink: 0,
          width: '200px',
          boxSizing: 'border-box',
          marginX: '2px',
          marginY: theme.spacing(2),
          marginBottom: 0,
        }}
        onMouseLeave={() => setHover(false)}
      >
```

File: frontend/src/components/Scenario/GroupFilterAppendage.tsx [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+        flexDirection: 'row',
+        marginY: theme.spacing(2),
+        marginLeft: '-3px',
+        padding: '1px',
+        height: 'min-content',
+        border: `1px solid ${props.color}`,
+        borderRadius: '3px',
+        background: theme.palette.background.paper,
+      }}
+    >
```
Comment (NXXR) [inline frontend/src/components/Scenario/GroupFilterAppendage.tsx:53]:
```suggestion
    <Box
      id={`scenario-card-${props.scenarioId}-group-filter-container`}
      sx={{
        display: 'flex',
        flexDirection: 'row',
        marginY: theme.spacing(2),
        marginLeft: '-3px',
        padding: '1px',
        height: 'min-content',
        border: `1px solid ${props.color}`,
        borderRadius: '3px',
        background: theme.palette.background.paper,
      }}
    >
```

File: frontend/src/components/Scenario/GroupFilterAppendage.tsx [hunk 8]
```diff
@@ -0,0 +1,88 @@
...
+      <Button
+        sx={{
+          width: '26px',
+          minWidth: '26px',
+          borderRight: `1px solid ${theme.palette.divider}`,
+        }}
+        onClick={() => setFolded(!folded)}
+      >
```
Comment (NXXR) [inline frontend/src/components/Scenario/GroupFilterAppendage.tsx:61]:
```suggestion
      <Button
        id={`scenario-card-${props.scenarioId}-group-filter-fold-toggle`}
        sx={{
          width: '26px',
          minWidth: '26px',
          borderRight: `1px solid ${theme.palette.divider}`,
        }}
        onClick={() => setFolded(!folded)}
      >
```

**Review by NXXR — CHANGES_REQUESTED** [review]

**Review by NXXR — APPROVED** [review]

**Pull request merged**